lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -461,6 +462,7 @@ def start(self):
                 background = self.cuts.apply_global_gh_cut(background)
 
             background = self.event_sel.filter_cut(background)
+            background = self.event_sel.same_disp_sign_filter(background)
```

Comment:
I don't understand what is the goal of this.

The only case in which we may want to apply the selection "only the events in which the head-tail estimate was correct" is when dealing with gammas, and for the sole purpose of finding an efficiency-based theta cut (or gammaness cut). The idea is to keep after the cut a given fraction of events in "the main peak of the PSF". Of course, in real data we don't know which events have right or wrong head-tail assignment, but it does not matter because (from the MC) we just obtain theta (and gammaness) cut values to be applied to those real data.

In no case we want to apply that "correct head-tail" cut  to the background, because what would it mean anyway? The head-tail finding is trained for gammas, and for most of the background events the outcome of the disp_sign RF will be more or less random, not related to the actual proton direction.

Reply:
I did not consider the implication of this selection on the background properly and had just proceeded to keep the event selection uniform. I will correct this.

Reply:
>. I am not sure why you would not want the IRFs to be created with the MC gamma events with the same disp_sign reconstruction. can you please further clarify on this?

IRFs (the actual IRFs, edisp, aeff and psf) must be computed on exactly the cuts you perform on real data. You must not apply any other cuts. It is ok to remove the wrong disp sign events for the optimization of cuts, but the calculation on IRFs must happen on all events.

Reply:
Ok! I understand now, thanks for the clarification. I will remove the changes made in the IRF Tool, and also in the evaluation of energy-dependent theta cuts function, but keep the `same_disp_sign_filter` function.